src/researchhub_document/permissions.py

```diff
@@ -1,29 +1,52 @@
-from utils.permissions import AuthorizationBasedPermission
-from utils.http import PATCH, POST, PUT, GET, DELETE
+from django.contrib.contenttypes.models import ContentType
+
+from hub.models import Hub
+from hypothesis.related_models.hypothesis import Hypothesis
+from paper.models import Paper
+from researchhub_access_group.constants import EDITOR
 from researchhub_document.related_models.constants.document_type import (
-    HYPOTHESIS,
     DISCUSSION,
+    HYPOTHESIS,
+    PAPER,
 )
-from hypothesis.related_models.hypothesis import Hypothesis
 from researchhub_document.models import (
     ResearchhubPost
 )
+from researchhub_document.related_models.researchhub_unified_document_model import ResearchhubUnifiedDocument
+from utils.permissions import AuthorizationBasedPermission
+
 
 class HasDocumentCensorPermission(AuthorizationBasedPermission):
     message = 'Need to be author or moderator to delete'
 
     def is_authorized(self, request, view, obj):
         if request.user.is_authenticated is False:
             return False
+        
+        document_type = obj.document_type if obj.__class__ is \
+            ResearchhubUnifiedDocument else Paper
+        doc = None
+        
+        if document_type == DISCUSSION:
+            doc = ResearchhubPost.objects.get(unified_document_id=obj.id)
+        elif document_type == HYPOTHESIS:
+            doc = Hypothesis.objects.get(unified_document_id=obj.id)
+        else:
+            doc = Paper.objects.get(id=obj.id)
```

Comment:
Instead of querying the database again for the object, we can just use the passed obj parameter.

Ex:
```
doc = obj
obj.get_document() # Create a class method to retrieve the document

# Class Method
if self.document_type == 'PAPER':
    doc = self.paper
elif ...
```

Reply:
your suggestion was what was there before.
I made changes like this because when a request was made against a paper, it comes in as a paper instance not a unidoc which means the "object" doesn't have a field document_type, resulting in an error. 
If i have a try-catch (which i also tried, the code looks messy) & if i had some sort of a check to filter it out, the code becomes redundant.